src/azure-cli/azure/cli/command_modules/appconfig/_kv_export_helpers.py

```diff
@@ -0,0 +1,441 @@
+                        )
+                    )
+                continue
+            previous_kv = KeyValue(key, kv.value)
+
+            # No need to construct for properties format
+            if format_ == "properties" or separator is None or not separator:
+                exported_dict.update({key: kv.value})
+                continue
+
+            key_segments = key.split(separator)
+            __export_keyvalue(key_segments, kv.value, exported_dict)
+
+        return __try_convert_to_arrays(exported_dict)
+    except Exception as exception:
+        raise CLIError("Fail to export key-values." + str(exception))
+
+
+def __export_features(retrieved_features, naming_convention, compatibility_mode):
+    new_ms_featuremanagement_keyword = (
+        FeatureManagementReservedKeywords.UNDERSCORE.feature_management
+    )
+    feature_flags_keyword = FeatureFlagConstants.FEATURE_FLAGS_KEY
+    try:
+        compatibility_mode = str(compatibility_mode).lower() in ["true"]
+        if compatibility_mode:
+            feature_reserved_keywords = FeatureManagementReservedKeywords.get_keywords(
+                naming_convention
+            )
+            exported_dict = {}
+
+            for feature in retrieved_features:
+                if (
+                    feature.allocation is not None or
+                    feature.variants is not None or
+                    feature.telemetry is not None
+                ):
+                    if new_ms_featuremanagement_keyword not in exported_dict:
+                        exported_dict[new_ms_featuremanagement_keyword] = {
+                            feature_flags_keyword: []
+                        }
+
+                    feature_entry = __export_feature_to_new_ms_schema(feature)
+                    exported_dict[new_ms_featuremanagement_keyword][
+                        feature_flags_keyword
+                    ].append(feature_entry)
+                else:
+                    if (
+                        feature_reserved_keywords.feature_management not in exported_dict
+                    ):
+                        exported_dict[feature_reserved_keywords.feature_management] = {}
+
+                    feature_entry = __export_feature_to_legacy_schema(
+                        feature, naming_convention
+                    )
+                    exported_dict[feature_reserved_keywords.feature_management].update(
+                        feature_entry
+                    )
+        else:
+            exported_dict = {
+                new_ms_featuremanagement_keyword: {feature_flags_keyword: []}
+            }
+            # retrieved_features is a list of FeatureFlag objects
+            for featureflag in retrieved_features:
+                feature_entry = __export_feature_to_new_ms_schema(featureflag)
+                exported_dict[new_ms_featuremanagement_keyword][
+                    feature_flags_keyword
+                ].append(feature_entry)
+
```

Comment:
I would prefer to keep it separate because we would get different kind of schemas when `compatibility_mode `is set to True. For example if the naming convention is underscore then the legacy schema would be nested within the new schema.